src/components/ShoppingList/ShoppingList.js

```diff
@@ -7,6 +8,21 @@ function ShoppingList({ listId }) {
     db.collection(`lists/${listId}/items`).orderBy('purchaseInterval', 'asc'),
   );
 
+  const handleCheck = (e) => {
+    if (!e.target.checked) return; // don't do anything if the checkbox isn't checked
+
+    const itemId = e.target.value;
+    db.collection(`lists/${listId}/items`)
+      .doc(itemId)
+      .update({
+        lastPurchaseDate: firebase.firestore.FieldValue.serverTimestamp(),
+        numberOfPurchases: firebase.firestore.FieldValue.increment(1),
+      })
```

Comment:
I like the .update here!

Reply:
+1 on the `.update` Excellent choice in using firebase functions to update these values.